cf-check/diagnose.c

```diff
@@ -291,8 +294,14 @@ static int lmdump_errno_to_code(int r)
 
 static int diagnose(const char *path)
 {
+    int saved_stdout = dup(STDOUT_FILENO);
     freopen("/dev/null", "w", stdout);
-    return lmdump(LMDUMP_VALUES_ASCII, path);
+    int ret = lmdump(LMDUMP_VALUES_ASCII, path);
+
+    char buf[32];
+    snprintf(buf, sizeof(buf), "/dev/fd/%d", saved_stdout);
+    freopen(buf, "w", stdout);
```

Comment:
Are you sure `/dev/fd/` exists on all the platforms where we support `cf-check diagnose`? There's also `/dev/stdout` on GNU/Linux.

Reply:
No, I can open a ticket to investigate this.